---
author: Sravani Vippaturi
output:
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    dev: 'svg'
  md_document:
    variant: gfm
bibliography: bibliography.ris
---

###Differential Expression Analysis

##Methods

1. Salmon is a tool for analyzing transcript abundance from RNA-seq data. You need a set of target transcripts (either from a reference assembly or a de-novo assembly) to use Salmon. You only need a FASTA file containing your reference transcripts and a set of FASTA files containing your de-novo transcripts. As described above, Salmon uses a mapping-based method: indexing followed by quantification. The indexing step is independent of the reads and only needs to be performed once for a particular set of reference transcripts. First method of operation is by building an index for the transcriptone and directly process subsequent reads. Secomd method of operation require .sam or .bam file with set of alignments within FASTA file of transcriptome.[1]

We build and indexed using salmon tool. Script file:
```{bash}
#!/usr/bin/env bash
# buildIndex.sh
salmon index -t /scratch/SampleDataFiles/Trinity.fasta -i AipIndex k 25
```

2. Aligned all Aip index files from /scratch/SampleDataFiles/Paired folder to an separate Quant folder in the working directory with help of below script:

```{bash}
#!/usr/bin/env bash
outDir='quant/'
sample='Aip02'
function align {
    salmon quant -l IU is \
        -1 /scratch/SampleDataFiles/Paired/$sample.R1.paired.fastq \
        -2 /scratch/SampleDataFiles/Paired/$sample.R2.paired.fastq \
        -i AipIndex \
        --validateMappings \
        -o $outDir$sample
}

align 1>align.log 2>align.err &
```

3. A transcript-level estimation program, tximport, imports transcript abundances, counts, and transcript lengths from external software and summarizes them at the gene-level (by default) or outputs transcript-level matrices. Before providing input to salmon transcripts are mapped to genes in tables. Created R-script mergeKo.R to build tables based on Annotation files in path /scratch/SampleDataFiles/Paired/scratch/SampleDataFiles/Annotation. to display in an organized manner we used kable() function in the below R-script. Performeed multiple tasks such as converting genes in to data frames, set solumn names, filter based on required conditions and merged tables with identical columns.

```{R, code=readLines("mergeKo.R"), echo=TRUE, eval=TRUE}
```


4. To import salmon alignments in to DESeq2 sequence and perform difeerential expression analysis we created an R-script using the Sample.csv from working directory.

```{r}
#!/usr/bin/env Rscript
# de.R

library(tximport)
library(readr)
library(DESeq2)
library(knitr)
tx2gene <- read.csv("tx2gene.csv")
head(tx2gene)
samples <- read.csv("Samples.csv", header=TRUE)
head(samples)
files <- file.path("quant", samples$Sample, "quant.sf")
txi <- tximport(files, type="salmon", tx2gene=tx2gene)
dds <- DESeqDataSetFromTximport(txi, colData = samples, design = ~ Menthol + Vibrio)
dds$Vibrio <- relevel(dds$Vibrio, ref = "Control")
dds$Menthol <- relevel(dds$Menthol, ref = "Control")
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]
dds <- DESeq(dds)
padj <- .05
minLog2FoldChange <- .5
dfAll <- data.frame()
# Get all DE results except Intercept, and "flatten" into a single file.
for (result in resultsNames(dds)){
    if(result != 'Intercept'){
        res <- results(dds, alpha=.05, name=result)
        dfRes <- as.data.frame(res)
        dfRes <- subset(subset(dfRes, select=c(log2FoldChange, padj)))
        dfRes$Factor <- result
        dfAll <- rbind(dfAll, dfRes)
    }
}

dfAll_Filter <- subset(dfAll, padj<0.05)
dfAll_Filter1 <- data.frame(dfAll_Filter, row.names=NULL)
dfAll_bind <- cbind(rownames(dfAll_Filter),dfAll_Filter1)
colnames(dfAll_bind)[1] <- "ko"
kable(dfAll_bind)

P_File <- "/home/vippaturi.s/BINF6309/module04-vippaturi-s-1/path.txt"
Path_ways <- read.table(P_File, sep="\t", header=FALSE)
colnames(Path_ways) <- c("ko", "Pathways")
ko_File <- "/home/vippaturi.s/BINF6309/module04-vippaturi-s-1/ko"
pathway <- read.table(ko_File, sep="\t", header=FALSE)
colnames(pathway) <- c("Pathways", "Description")
df_Pathway <- merge(Path_ways, pathway)
deAnnotated <- merge(df_Pathway, dfAll_bind)
kable(deAnnotated)
write.csv(deAnnotated, file="deAnnotated.csv")
```
##Results

Results are populated along with Script files.

##References

[1]Roberts, Adam, and Lior Pachter. “Streaming fragment assignment for real-time analysis of sequencing experiments.” Nature Methods 10.1 (2013): 71-73.


